docs/design/autoware-interfaces/component/.pages

```diff
@@ -0,0 +1,3 @@
+nav:
+  - index.md
+  - Architecture: planning-control-vehicle-interface.md
```

Comment:
Why did you customize the title here?

Reply:
Because it is too long for menu.

Reply:
I see, but writing just `Architecture` is ambiguous to me. :thinking:

Reply:
So how about splitting this file into `planning.md`, `control.md`, and `vehicle-interface.md`?

cc @TakaHoribe

Reply:
Also, creating empty files for other components would be good.

Reply:
@isamu-takagi cc @TakaHoribe ping.

Reply:
@kenji-miyake @TakaHoribe 
Split the component architecture page. How about this? https://github.com/autowarefoundation/autoware-documentation/pull/57/commits/0308758a166568dd6f93cbee0e474e1cab82b72a